transformer_engine/pytorch/attention.py

```diff
@@ -2134,6 +2135,13 @@ def forward(
         if not _flash_attn_2_available and self.num_gqa_groups != self.num_attention_heads:
             use_flash_attention = False
 
+        # Disable FAv2.1+ for causal mask in cross attention due to
+        # https://github.com/Dao-AILab/flash-attention#21-change-behavior-of-causal-flag
+        if (_flash_attn_2_1_plus
+            and attn_mask_type == "causal"
+            and max_seqlen_q != max_seqlen_kv):
+            use_flash_attention = False
```

Comment:
I think a warning would be useful here pointing to the change in behavior in FA's README.

Reply:
Sure